### ISSUE TITLE
Checkbox/radio buttons for exposed filters in views

### ISSUE THREAD

**Issue by richard-walsh** [govCMS/GovCMS7#403]
It would be great to be able to use checkboxes and radio buttons in exposed filters in views. Currently, the only options are Dropdown and autocomplete, which function fine but do not provide the best user experience. A module similar to https://www.drupal.org/project/better_exposed_filters could provide such functionality.

**Comment by aleayr** [comment]
Hi Richard,

Thanks for the idea. Do you have some specific use cases this would be beneficial for?

**Comment by richard-walsh** [comment]
I think it would be most beneficial in scenarios where multiple filters can be selected at once. (Anywhere you would currently use a dropdown list with multiple selections allowed.)
From a users perspective I think checkboxes are more intuitive than using ctrl+click (or equivalent) to select multiple items.
The specific case that I'm looking to solve is recreating this (http://www.greatstart.sa.edu.au/Activity.aspx?cat=6) filtering system in govcms.
Since my original post I have realised that facetapi could also provide what I am looking for, but I haven't looked into the specifics of how it works yet.

**Comment by nathan-w** [comment]
I agree - check boxes are more intuitive than long select lists, and multi-select lists are just an abomination... :) 

We also need to consider placement and styling of the exposed filters, regardless of what input type they are. Using a consistent design pattern would be a good thing. And what would happen if the exposed filter contained a long long looooooong list of check boxes? Scrolling through overflow is also a bit messy, but a really long list of check boxes will also present problems.

Be good to see some user research around how well the concept of 'filtering' works. We may need to provide alternative mechanisms of navigation...

**Comment by richard-walsh** [comment]
When it comes to long lists of check boxes sometimes it is appropriate and effective to list them horizontally first (and then rows). This works quite well as long as the labels are of a similar length so that you don't lose any information.

This is an example of how I chose to layout a list of checkboxes in the backend of our site:
![skills-list](https://cloud.githubusercontent.com/assets/22208191/22192332/15030554-e183-11e6-9632-0b4787880fbe.png)

You can see that some of the longer labels are cutoff, and that if you don't know what information you are looking for it is a little unwieldy. But in the situation it was designed for, it works perfectly as the user (editor/publisher) already knows which specific tickboxes need to be selected.

**Comment by DJMyles** [comment]
I've picked this one up as part of the camp since we actually use BEF (better exposed forms) on our live sight at the moment to filter our forms - https://www.humanservices.gov.au/health-professionals/forms/help-me-find-a-form 

In terms of coming up with a solution to a long taxonomy of checkboxes, our argument would be that this becomes a content, or more broadly an IA issue, rather than something a technical solution should attempt to solve. For example, if I want to filter by say an audience taxonomy which contains 98 different terms, as a content owner I may wish to consider addressing why I have so many audience types and look at refining or looking at categorising them into top and sub level categories in a taxonomy hierarchy, which in itself would be more wieldy forming part of filtering an exposed form. 

The added benefit of this approach is that it not only addresses potential bloat in the site structure, but also likely simplifies the end user experience.

**Comment by alex-page** [comment]
I thought this was a default SaaS module! Here is a PR #481

BEF greatly enhances the default exposed filters and gives more options for accessibility. Allowing admins to customise labels, add descriptions and choose the appropriate display of the filters.

It also increases user experience allowing easy selection of multiple filters and the option to clear all filters. It also gives the current selected filters an active state which allows for different styling.
